test/models/order_test.rb

```diff
@@ -36,4 +36,34 @@ class OrderTest < ActiveSupport::TestCase
 
     assert_not order.make_placed
   end
-end
+
+  test 'not enough inventory' do
+    order = create(:order)
+    product = create(:product)
+    cc = create(:credit_card)
+    order.user.credit_card = cc
+    order.save
+
+    order_item = OrderItem.create(order_id: order.id, source_type: Product.name, source_id: product.id, quantity: 15)
```

Comment:
it would just be an empty factory since all the fields are getting set here, right?